### PR TITLE
fix: clean up replenishment [pre-mycompany-merge] (GH-8901)

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes GH-8901

**This clean up is intended for pre-my-company merge

A separate clean up for replenishment will be done in another ticket regarding the post-merge of mycompany into develop**

**clean up includes**

- fixing hardcoded terms and checked for placing a singular order (was like that already)
- change the deprecated way of using `makeErrorSerializable` to `normalizeHttpError` on files that replenishment touched
- fix endpoints for fields in relation to anything replenishment touched and placing it in the configurable endpoints (instead of setting the fields in adapter)
- clean up occ model and replenishment model

### CODE REVIEW

File: projects/core/src/occ/adapters/checkout/default-occ-checkout-config.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export const defaultOccCheckoutConfig: OccConfig = {
       endpoints: {
         setDeliveryAddress:
           'users/${userId}/carts/${cartId}/addresses/delivery',
-        placeOrder: 'users/${userId}/orders',
+        placeOrder: 'users/${userId}/orders?fields=FULL',
```
Comment (LTiger14) [inline projects/core/src/occ/adapters/checkout/default-occ-checkout-config.ts:9]: Can you be more specific?

File: projects/storefrontlib/src/recipes/config/b2b-occ-config.ts [hunk 2]
```diff
@@ -8,9 +8,9 @@ export const defaultB2bOccConfig: OccConfig = {
         addEntries: 'orgUsers/${userId}/carts/${cartId}/entries',
         setDeliveryAddress:
           'orgUsers/${userId}/carts/${cartId}/addresses/delivery',
-        // TODO(#8877): Is this a hack for an API? Shouldn't user pass this data?
-        placeOrder: 'orgUsers/${userId}/orders?termsChecked=true',
-        scheduleReplenishmentOrder: 'orgUsers/${userId}/replenishmentOrders',
+        placeOrder: 'orgUsers/${userId}/orders?fields=FULL',
```
Comment (LTiger14) [inline projects/storefrontlib/src/recipes/config/b2b-occ-config.ts:11]: Do we always need full?

File: projects/storefrontlib/src/cms-components/myaccount/order/replenishment-order-details/replenishment-order-cancellation/replenishment-order-cancellation.component.html [hunk 3]
```diff
@@ -15,7 +15,7 @@
     *ngIf="(replenishmentOrder$ | async).active"
     class="col-xs-12 col-md-4 col-lg-3"
   >
-    <button class="btn btn-block btn-action" (click)="openDialog()">
+    <button #element class="btn btn-block btn-action" (click)="openDialog()">
```
Comment (LTiger14) [inline projects/storefrontlib/src/cms-components/myaccount/order/replenishment-order-details/replenishment-order-cancellation/replenishment-order-cancellation.component.html:18]: What is this id?

Comment (bgambocjaviniar) [reply projects/storefrontlib/src/cms-components/myaccount/order/replenishment-order-details/replenishment-order-cancellation/replenishment-order-cancellation.component.html:18]: it was for the view child, which wasn't hooked up before in the component (the one similar to anonymous) and that will be sent when calling the .openDialog function

File: projects/storefrontlib/src/recipes/config/b2b-occ-config.ts (hunk 2 above)
Comment (bgambocjaviniar) [reply projects/storefrontlib/src/recipes/config/b2b-occ-config.ts:11]: this was placed in the adapter before and I just moved it out to the configurable endpoint

File: projects/core/src/occ/adapters/checkout/default-occ-checkout-config.ts (hunk 1 above)
Comment (bgambocjaviniar) [reply projects/core/src/occ/adapters/checkout/default-occ-checkout-config.ts:9]: answered below, but it was already there and I don't know which exact properties it needs but the properties are the ones we need to populate the products. I believe it's best to keep it as how it was done before

**Review by LTiger14 — APPROVED** [review]

**Pull request merged**